lib/tilex/rate_limiter.ex

```diff
@@ -0,0 +1,46 @@
+defmodule Tilex.RateLimiter do
+  use GenServer
+
+  @name __MODULE__
+  @date_time_module Application.get_env(:tilex, :date_time_module)
+  @limit Application.get_env(:tilex, :rate_limiter_requests_per_time_period)
+  @time_period_minutes Application.get_env(:tilex, :rate_limiter_time_period_minutes)
+
+  # Client ----------------------------------------------
+
+  def start_link do
+    GenServer.start_link(__MODULE__, [], name: @name)
+  end
+
+  def check(ip: ip) do
+    GenServer.call(pid(), {:check_ip, ip})
```

Comment:
At this point I have less Elixir experience than both of you, but is there a reason to use `Process.whereis/1` if we are using a named server (ie: `@name`)? Is there any benefit to this over just calling `GenServer.call(@name, {:check_ip, ip})`? (I have a feeling I'm about to learn something new by asking this 😄 )

Reply:
No I think you're right, you can just use `GenServer.call(@name, message)`.